release/RELEASE_PROCESS.md

```diff
@@ -2,48 +2,162 @@
-     - BSD tar: `RELEASE=update-me;tar -zcvf ${RELEASE}.tar.gz -s /config/${RELEASE}/ config/`
-1. Once everything has settled, tag and push the release with `git tag $VERSION` and `git push upstream $VERSION`. You can also tag the release in the GitHub UI.
-   - The `modelmesh-controller` image will be published via GitHub Actions.
-1. Upload generated install manifests to GitHub release assets.
-1. Be sure to create and push a new tag for the `kserve/modelmesh-minio-examples` image.
+It's generally a good idea to search the repo or `control`-`shift`-`f` for strings
+of the old version number and replace them with the new, keeping in mind conflicts
+with other library version numbers.
+
+Some of the steps below need to be performed at least twice:
+
+- once for the release candidate(s) (`v0.10.0-rc0`, `v0.10.0-rc1`, ...) and
+- once more for the actual release (`v0.10.0`).
+
+1. Create new (pre-)release tags (v...`-rc0`) in these repositories:
+
+   - [ ] [`modelmesh`](https://github.com/kserve/modelmesh/releases)
+   - [ ] [`modelmesh-runtime-adapter`](https://github.com/kserve/modelmesh-runtime-adapter/releases)
+   - [ ] [`rest-proxy`](https://github.com/kserve/rest-proxy/releases)
+
+   using the newly created `release-*` branches as target. This can be done by
+   creating a draft release using the GitHub web interface and checking the
+   "Set as a pre-release" option. The newly created tag should trigger the GitHub
+   action to push the respective (pre-)release container images to DockerHub which
+   are needed in the next step.
+
+2. Build and push a new tag for the `kserve/modelmesh-minio-examples` image. If
+   there have been no changes since the last release, you can create a new tag
+   from the previous release, e.g. creating a new `v0.10.0` tag from `latest`:
+
+   ```shell
+   docker manifest create kserve/modelmesh-minio-examples:v0.10.0 kserve/modelmesh-minio-examples:latest
+   docker manifest push kserve/modelmesh-minio-examples:v0.10.0
+   ```
+
+3. Verify image tags got pushed to [DockerHub](https://hub.docker.com/u/kserve):
+
+   - [ ] [kserve/modelmesh](https://hub.docker.com/r/kserve/modelmesh/tags)
+   - [ ] [kserve/modelmesh-minio-examples](https://hub.docker.com/r/kserve/modelmesh-minio-examples/tags)
+   - [ ] [kserve/modelmesh-runtime-adapter](https://hub.docker.com/r/kserve/modelmesh-runtime-adapter/tags)
+   - [ ] [kserve/rest-proxy](https://hub.docker.com/r/kserve/rest-proxy/tags)
+
+4. In this `modelmesh-serving` repository, update the container image tags to
+   the corresponding release versions for:
+
+   - `kserve/modelmesh`
+   - `kserve/modelmesh-controller`
+   - `kserve/modelmesh-minio-examples`
+   - `kserve/modelmesh-runtime-adapter`
+   - `kserve/rest-proxy`
+
+   The version tags should be updated in the following files:
+
+   - [ ] `config/default/config-defaults.yaml`: edit the image tags for ...
```

Comment:
Nit: I like the helpful component checklist but could use "edit each component's image tags" to remove the `...`